rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/UsersRouter.scala

```diff
@@ -31,48 +34,80 @@ import com.actionml.circe.CirceSupport
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
 class UsersRouter(implicit injector: Injector) extends Directives with Injectable with CirceSupport with AuthorizationDirectives {
   override val authorizationService = inject[AuthorizationService]
   private val config = inject[AppConfig]
   override val authEnabled = config.authServer.authorizationEnabled
 
-  def route: Route = (handleExceptions(exceptionHandler) & extractLog) { implicit log =>
-    (pathPrefix("auth" / "users") & extractAccessToken) { implicit token =>
-      (pathEndOrSingleSlash & post & hasAccess(user.create) & entity(as[CreateUserRequest])) {
-        case CreateUserRequest(roleSetId, resourceId) =>
-          onSuccess(usersService.create(roleSetId, resourceId))(complete(_))
+  def route: Route = extractLog { implicit log =>
+    (pathPrefix("auth" / "users") & handleExceptions(exceptionHandler(log)) & extractAccessToken) { implicit token =>
+      pathEndOrSingleSlash {
+        (get & parameters('offset.as[Int] ? 0, 'limit.as[Int] ? Int.MaxValue) & hasAccess(user.permissions)) { (offset, limit) =>
+          onSuccess(usersService.list(offset = offset, limit = limit))(complete(_))
+        } ~
+        (post & entity(as[CreateUserRequest]) & hasAccess(user.create)) {
+          case CreateUserRequest(roleSetId, resourceId) =>
+            onSuccess(usersService.create(roleSetId, resourceId.getOrElse(ResourceId.*)))(complete(_))
+        }
       } ~
-      (path(Segment / "permissions") & hasAccess(user.permissions)) { userId =>
-        (post & entity(as[PermissionsRequest])) { case PermissionsRequest(roleSetId, resourceId) =>
-          onSuccess(usersService.grantPermissions(userId, roleSetId, resourceId)) { _ =>
-            complete(PermissionsResponse(userId, roleSetId, resourceId))
-          }
+      (pathPrefix(Segment) & hasAccess(user.permissions)) { userId =>
+        pathEndOrSingleSlash {
+          get { getUser(userId) } ~
+          delete { deleteUser(userId) }
         } ~
-        (delete & parameter('roleSetId)) { roleSetId =>
-          onSuccess(usersService.revokePermissions(userId, roleSetId)) { _ =>
-            complete(PermissionsResponse(userId, roleSetId, ResourceId.*))
+        path("permissions") {
+          (post & entity(as[PermissionsRequest])) { req =>
+            val resourceId = req.resourceId.getOrElse(ResourceId.*)
+            onSuccess(usersService.grantPermissions(userId, req.roleSetId, resourceId)) {
+              complete(PermissionsResponse(userId, req.roleSetId, resourceId))
+            }
+          } ~
+          (delete & parameters('roleSetId, 'resourceId ?)) { (roleSetId, resourceIdOpt) =>
+            val resourceId = resourceIdOpt.getOrElse(ResourceId.*)
+            onSuccess(usersService.revokePermissions(userId, roleSetId, resourceId)) {
+              complete(PermissionsResponse(userId, roleSetId, resourceId))
+            }
           }
```

Comment:
Please document REST API changes in rest_spec.md

Reply:
I added offset and limit (default is 0 and MAX_INT), and all other changes were made to fit rest_spec